launch.sh

```diff
@@ -8,88 +8,126 @@
 set -e
 
 die() {
-	echo "ci configuration failed: $@" 1>&2
+	echo "CI internal failure : $@" 1>&2
 	exit 1
 }
 
 # Standard help message
 usage()
 {
     cat <<EOF
+    This script is the main launcher for the SEAPATH CI.
+    It is separated in three functions in order to display logs properly.
+    They should be called one after another.
 USAGE:
-    ./launch.sh <pull-request-refs> <pull-request-sha>
+    ./launch.sh <config|setup|test>
 EOF
 }
 
-if test "$#" -ne 2; then
-  usage
-  die "invalid usage"
-fi
+# Download and prepare the pull request sources
+initialization() {
+  if ! type -p cqfd > /dev/null; then
+    die "cqfd not found"
+  fi
 
-# Prerequisite
-if ! type -p cqfd > /dev/null; then
-  die "cqfd not found"
-fi
+  # Get sources
+  git clone -q https://github.com/seapath/ansible
+  cd ansible
+  git fetch -q origin ${GITHUB_REF}
+  git checkout -q FETCH_HEAD
+  echo "Pull request sources got succesfully"
 
-PR_BRANCH=$1
-PR_HASH=`echo $2 | cut -b 1-7`
-CI_DIR=`pwd`
-PR_N=`echo $PR_BRANCH | cut -d '/' -f 3`
-TIME=`date +%Hh%Mm%S`
-CUKINIA_TEST_DIR=$CI_DIR/cukinia_tests
-REPORT_NAME=test-report_${PR_HASH}_${TIME}.pdf
-REPORT_DIR=$CI_DIR/reports/docs/reports/PR-${PR_N}
+  # Prepare ansible repository
+  cqfd init
+  cqfd -b prepare
+  echo "Sources prepared succesfully"
+}
+
+# Launch Debian configuration and hardening
+configure_debian() {
+  cd ansible
+  LOCAL_ANSIBLE_DIR=/home/virtu/ansible # Local dir that contains keys and inventories
+  CQFD_EXTRA_RUN_ARGS="-v $LOCAL_ANSIBLE_DIR:/tmp" cqfd run ansible-playbook \
```

Comment:
Mounting bind the LOCAL_ANSIBLE_DIR into /tmp is not a good idea because other software can also write into /tmp. You can use a subdirectory in /tmp/ instead.